### PR TITLE
Create set_name endpoint

### PR DESCRIPTION
**Pull request by rossdanlm**
Create set_name endpoint

TSIA

## Test Plan
Follow dev README to setup the local editor: https://github.com/lastmile-ai/aiconfig/tree/main/python/src/aiconfig/editor#dev, then run this command
```
curl http://localhost:8080/api/set_name -d '{"name":"cool_new_name"}' -X POST -H 'Content-Type: application/json'
```

https://github.com/lastmile-ai/aiconfig/assets/151060367/efcb7ceb-e390-4208-93b3-3effd5d66f8b

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/lastmile-ai/aiconfig/pull/692).
* #693
* __->__ #692
* #691
* #690
* #688
* #670
* #668

### CODE REVIEW

File: python/src/aiconfig/editor/server/server.py [hunk 1]
```diff
@@ -269,7 +270,18 @@ def delete_parameter() -> FlaskResponse:
     prompt_name: str | None = request_json.get("prompt_name")
 
     operation = make_op_run_method(MethodName("delete_parameter"))
-    operation_args: Result[OpArgs, str] = result.Ok(
-        OpArgs({"parameter_name": parameter_name, "prompt_name": prompt_name})
-    )
+    operation_args: Result[OpArgs, str] = result.Ok(OpArgs({"parameter_name": parameter_name, "prompt_name": prompt_name}))
     return run_aiconfig_operation_with_op_args(aiconfig, "delete_parameter", operation, operation_args)
```
Comment (jonathanlastmileai) [inline python/src/aiconfig/editor/server/server.py:274]: Is this auto-linting? Can you try to do that atomically for large sets of files in their own PRs, or even better do it before submitting the PR that modifies that specific code?

File: python/src/aiconfig/editor/server/server.py [hunk 2]
```diff
@@ -269,7 +270,18 @@ def delete_parameter() -> FlaskResponse:
...
+def set_name() -> FlaskResponse:
+    state = get_server_state(app)
+    aiconfig = state.aiconfig
+    request_json = request.get_json()
+
+    name: str | None = request_json.get("name")
+
+    operation = make_op_run_method(MethodName("set_name"))
+    operation_args: Result[OpArgs, str] = result.Ok(OpArgs({"name": name}))
+    return run_aiconfig_operation_with_op_args(aiconfig, "set_name", operation, operation_args)
```
Comment (jonathanlastmileai) [inline python/src/aiconfig/editor/server/server.py:287]: I'm fairly sure you can use the wrapper function here (see a previous comment). If not, at this point I would probably refactor a bit to make it usable. I usually use the rule of thumb that a set of 3 cases justifies an abstraction. 

If we hadn't done that to define `make_op_run_method()` and `run_aiconfig_operation_with_op_args()` in the first place, we would have either a lot more code duplication than this or a lot less robust error handling. We can probably go a little further here.

**Review by rholinshead — APPROVED** [review]

File: python/src/aiconfig/editor/server/server.py (hunk 1 above)
Comment (rossdanlm) [reply python/src/aiconfig/editor/server/server.py:274]: You're right, my bad I forgot to do it last time

File: python/src/aiconfig/editor/server/server.py (hunk 2 above)
Comment (rossdanlm) [reply python/src/aiconfig/editor/server/server.py:287]: Yea you're right, I was just copy pasting stuff

**Pull request merged**